ribs/archives/_cvt_archive.py

```diff
@@ -164,16 +170,59 @@ def __init__(self,
         self._chunk_size = chunk_size
 
         if custom_centroids is None:
-            if not isinstance(samples, int):
-                # Validate shape of custom samples. These are ignored when
-                # `custom_centroids` is provided.
-                samples = np.asarray(samples, dtype=self.dtype)
-                if samples.shape[1] != self._measure_dim:
-                    raise ValueError(
-                        f"Samples has shape {samples.shape} but must be of "
-                        f"shape (n_samples, len(ranges)={self._measure_dim})")
-            self._samples = samples
-            self._centroids = None
+            if centroid_method == "kmeans":
+                # Samples are set to custom first and then checked or used.
+                self._samples = samples
+                if not isinstance(samples, int):
+                    # Validate shape of custom samples.
+                    samples = np.asarray(samples, dtype=self.dtype)
+                    if samples.shape[1] != self._measure_dim:
+                        raise ValueError(
+                            f"Samples has shape {samples.shape} but must be of "
+                            f"shape (n_samples, len(ranges)="
+                            f"{self._measure_dim})")
+                else:
+                    self._samples = self._rng.uniform(
+                        self._lower_bounds,
+                        self._upper_bounds,
+                        size=(self._samples, self._measure_dim),
+                    ).astype(self.dtype) if isinstance(self._samples,
```

Comment:
The if condition here is unnecessary since you just checked samples in the if statement.